app/pages/project/vpcs/VpcPage/tabs/VpcRoutersTab.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+import { createColumnHelper } from '@tanstack/react-table'
+import { useCallback, useMemo } from 'react'
+import { Outlet, useNavigate, type LoaderFunctionArgs } from 'react-router-dom'
+
+import { apiQueryClient, useApiMutation, type VpcRouter } from '@oxide/api'
+
+import { getVpcSelector, useVpcSelector } from '~/hooks'
+import { confirmDelete } from '~/stores/confirm-delete'
+import { addToast } from '~/stores/toast'
+import { makeLinkCell } from '~/table/cells/LinkCell'
+import { useColsWithActions, type MenuAction } from '~/table/columns/action-col'
+import { Columns } from '~/table/columns/common'
+import { PAGE_SIZE, useQueryTable } from '~/table/QueryTable'
+import { CreateLink } from '~/ui/lib/CreateButton'
+import { EmptyMessage } from '~/ui/lib/EmptyMessage'
+import { pb } from '~/util/path-builder'
+
+const colHelper = createColumnHelper<VpcRouter>()
+
+VpcRoutersTab.loader = async ({ params }: LoaderFunctionArgs) => {
+  const { project, vpc } = getVpcSelector(params)
+  await apiQueryClient.prefetchQuery('vpcRouterList', {
+    query: { project, vpc, limit: PAGE_SIZE },
+  })
+  return null
+}
+
+export function VpcRoutersTab() {
+  const vpcSelector = useVpcSelector()
+  const navigate = useNavigate()
+  const { project, vpc } = vpcSelector
+  const { Table } = useQueryTable('vpcRouterList', {
+    query: { project, vpc, limit: PAGE_SIZE },
+  })
+
+  const emptyState = (
+    <EmptyMessage
+      title="No VPC routers"
+      body="Create a router to see it here"
+      buttonText="New router"
+      buttonTo={pb.vpcRoutersNew({ project, vpc })}
+    />
+  )
+
+  const staticColumns = useMemo(
+    () => [
+      colHelper.accessor('name', {
+        cell: makeLinkCell((router) => pb.vpcRouter({ ...vpcSelector, router })),
+      }),
+      colHelper.accessor('description', Columns.description),
+    ],
+    [vpcSelector]
+  )
```

Comment:
Table looks sparse. Might as well add a created time column like in the subnets table. That's our usual table padding.

![Screenshot 2024-08-13 at 12 07 26 PM](https://github.com/user-attachments/assets/9888146f-76f2-41cf-9738-d38d4d5bdf37)